src/parser/deserializer.ts

```diff
@@ -75,9 +75,7 @@ export function deserialize(
 
   if (size + index > buffer.length) {
     throw new Error(
-      `(bson size ${size} + options.index ${index} must be <= buffer length ${Buffer.byteLength(
-        buffer
-      )})`
+      `(bson size ${size} + options.index ${index} must be <= buffer length ${buffer.byteLength})`
```

Comment:
I think we should continue to depend on [Buffer.byteLength](https://nodejs.org/api/buffer.html#buffer_static_method_buffer_bytelength_string_encoding), what if this is a `Buffer` instance? I'm sure there's a reason they provide the helper even though `Buffer` is implemented as a `UInt8Array`

Reply:
> When string is a Buffer/DataView/TypedArray/ArrayBuffer/ SharedArrayBuffer, the byte length as reported by .byteLength is returned.

So this is just grabbing the same property. I changed this because the typings for this function only accept string even though it takes a variety of types 🤷  could cast to fix?

Reply:
oh I see, I missed that the method only accepts a string with an encoding. Move along!